mtenn/conversion_utils/visnet.py

```diff
@@ -0,0 +1,241 @@
+"""
+Representation and strategy for ViSNet model.
+"""
+from copy import deepcopy
+import torch
+from torch.autograd import grad
+from torch_geometric.utils import scatter
+
+from mtenn.model import GroupedModel, Model
+from mtenn.strategy import ComplexOnlyStrategy, ConcatStrategy, DeltaStrategy
+
+# guard required: currently require PyG nightly 2.5.0 (SEE ISSUE #42)
+HAS_VISNET = False
+
+try:
+    from torch_geometric.nn.models import ViSNet as PygVisNet
+    HAS_VISNET = True
+except ImportError:
+    pass
+
+
+class EquivariantVecToScalar(torch.nn.Module):
+    # Wrapper for PygVisNet.EquivariantScalar to implement forward() method
+    def __init__(self, mean, reduce_op):
+        super(EquivariantVecToScalar, self).__init__()
+        self.mean = mean
+        self.reduce_op = reduce_op
+    def forward(self, x):
+        # dummy variable. all atoms from the same molecule and the same batch
+        batch = torch.zeros(x.shape[0], dtype=torch.int64, device=x.device)
+
+        y = scatter(x, batch, dim=0, reduce=self.reduce_op)
+        return y + self.mean
+
+
+if HAS_VISNET:
+    class ViSNet(torch.nn.Module):
+        def __init__(self, *args, model=None, **kwargs):
+            super().__init__()
+            ## If no model is passed, construct default SchNet model, otherwise copy
+            ##  all parameters and weights over
+            if model is None:
+                self.visnet = PygVisNet(*args, **kwargs)
+            else:
+                try:
+                    atomref = model.atomref.weight.detach().clone()
+                except AttributeError:
+                    atomref = None
```

Comment:
Looks like they handle `atomref` a bit differently here than in SchNet. They also always have it defined, even if a `None` value is passed when constructing the object. I think it should look something like this:
```suggestion
                atomref = model.prior_model.atomref.weight.detach().clone()
```
(see https://pytorch-geometric.readthedocs.io/en/latest/_modules/torch_geometric/nn/models/visnet.html#ViSNet)